autopr/__init__.py

```diff
@@ -155,28 +160,46 @@ def test():
 
 
 @cli.command()
+@click.option(
+    "--pull-repos/--no-pull-repos",
+    default=True,
+    is_flag=True,
+    help="Whether to pull repositories before running",
+)
 @click.option(
     "--push-delay",
-    type=click.FLOAT,
-    default=None,
+    type=click.FloatRange(min=0.0, max=None, clamp=True),
+    default=DEFAULT_PUSH_DELAY,
     help="Delay in seconds between pushing changes to repositories",
 )
-def run(push_delay: Optional[float]):
+def run(pull_repos: bool, push_delay: Optional[float]):
     """ Run update logic and create pull requests if changes made """
     cfg = workdir.read_config(WORKDIR)
     db = workdir.read_database(WORKDIR)
     _ensure_set_up(cfg, db)
     gh = github.create_github_client(cfg.credentials.api_key)
 
-    for repository in db.non_removed_repositories():
-        if repository.done:
-            continue
+    repositories = db.non_removed_repositories()
+    repositories_todo = [r for r in repositories if not r.done]
```

Comment:
Could be a function on the Database

Reply:
Will add one once it can be reused. `non_removed_repositories` mostly exists because I kept forgetting to check if the repository was removed.